presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -601,7 +602,17 @@ protected Type visitLogicalBinaryExpression(LogicalBinaryExpression node, Stacka
         protected Type visitComparisonExpression(ComparisonExpression node, StackableAstVisitorContext<Context> context)
         {
             OperatorType operatorType = OperatorType.valueOf(node.getOperator().name());
-            return getOperator(context, node, operatorType, node.getLeft(), node.getRight());
+            Type outputType = getOperator(context, node, operatorType, node.getLeft(), node.getRight());
+            // this needs to be checked after the call to getOperator(), because that's where the argument types get analyzed
+            if (sqlFunctionProperties.shouldWarnOnCommonNanPatterns() &&
+                    (TypeUtils.isApproximateNumericType(getExpressionType(node.getLeft())) || TypeUtils.isApproximateNumericType(getExpressionType(node.getRight())))) {
+                warningCollector.add(new PrestoWarning(
+                        SEMANTIC_WARNING,
+                        "Semantics of comparisons involving NaNs are changing. Comparisons involving DOUBLE/REAL types may produce" +
```

Comment:
Can this be reworded to simply indicate that expressions involving doubles may produce NaNs?  i.e. not in the context of an evolving process, but as a simple statement of fact?

Reply:
How about something like: `Comparison operations involving DOUBLE or REAL types may include NaNs in the input. Consider filtering out NaN values from your comparison using the is_nan() function.`

Reply:
I've updated to this wording

Reply:
SGTM!